modules/sql/src/test/scala/SqlNestedEffectsMapping.scala

```diff
@@ -0,0 +1,248 @@
+  val CountryType  = schema.ref("Country")
+  val CityType     = schema.ref("City")
+  val LanguageType = schema.ref("Language")
+  val CurrencyType = schema.ref("Currency")
+
+  val typeMappings =
+    List(
+      ObjectMapping(
+        tpe = QueryType,
+        fieldMappings = List(
+          SqlObject("cities"),
+          SqlObject("country")
+        )
+      ),
+      ObjectMapping(
+        tpe = CountryType,
+        fieldMappings = List(
+          SqlField("code",            country.code, key = true, hidden = true),
+          SqlField("name",            country.name),
+          SqlField("continent",       country.continent),
+          SqlField("region",          country.region),
+          SqlField("surfacearea",     country.surfacearea),
+          SqlField("indepyear",       country.indepyear),
+          SqlField("population",      country.population),
+          SqlField("lifeexpectancy",  country.lifeexpectancy),
+          SqlField("gnp",             country.gnp),
+          SqlField("gnpold",          country.gnpold),
+          SqlField("localname",       country.localname),
+          SqlField("governmentform",  country.governmentform),
+          SqlField("headofstate",     country.headofstate),
+          SqlField("capitalId",       country.capitalId),
+          SqlField("code2",           country.code2),
+          SqlObject("cities",         Join(country.code, city.countrycode)),
+          SqlObject("languages",      Join(country.code, countrylanguage.countrycode)),
+          EffectMapping("currencies", CurrencyQueryHandler)
+        ),
+      ),
+      ObjectMapping(
+        tpe = CityType,
+        fieldMappings = List(
+          SqlField("id", city.id, key = true, hidden = true),
+          SqlField("countrycode", city.countrycode, hidden = true),
+          SqlField("name", city.name),
+          SqlField("district", city.district),
+          SqlField("population", city.population),
+          SqlObject("country", Join(city.countrycode, country.code)),
+        )
+      ),
+      ObjectMapping(
+        tpe = LanguageType,
+        fieldMappings = List(
+          SqlField("language", countrylanguage.language, key = true, associative = true),
+          SqlField("isOfficial", countrylanguage.isOfficial),
+          SqlField("percentage", countrylanguage.percentage),
+          SqlField("countrycode", countrylanguage.countrycode, hidden = true),
+          SqlObject("countries", Join(countrylanguage.countrycode, country.code))
+        )
+      )
+    )
+
+  object CurrencyQueryHandler extends EffectHandler[F] {
+    def runEffects(queries: List[(Query, Cursor)]): F[Result[List[(Query, Cursor)]]] = {
+      val countryCodes = queries.map(_._2.fieldAs[String]("code").toOption)
+      val distinctCodes = queries.flatMap(_._2.fieldAs[String]("code").toList).distinct
+
+      val children = queries.flatMap {
+        case (PossiblyRenamedSelect(Select(name, _, child), alias), parentCursor) =>
```

Comment:
`PossiblyRenamedSelect` is an extractor/constructor for a `Select` possibly wrapped in a `Rename` node so that they can be handled together conveniently. It's a good question whether it's actually needed here, or if instead just `Select` would do. I'll check, and simplify if possible.

Reply:
I've added a test which shows we do need this here. We need to be able to see the aliases in the handler so that we can differentiate aliased paths.